Admin/AbstractAdmin.php

```diff
@@ -1940,6 +1962,61 @@ public function getParent()
     }
 
     /**
+     * Return the root ancestor or itself if not a child.
+     *
+     * @return AdminInterface
+     */
+    final public function getRootAncestor()
+    {
+        $parent = $this;
+
+        while ($parent->isChild()) {
+            $parent = $parent->getParent();
+        }
+
+        return $parent;
+    }
+
+    /**
+     * Return the depth of the admin.
+     * e.g. 0 if not a child; 2 if child of a child; etc...
+     *
+     * @return int
+     */
+    final public function getChildDepth()
+    {
+        $parent = $this;
+        $depth = 0;
+
+        for (; $parent->isChild(); ++$depth) {
+            $parent = $parent->getParent();
+        }
+
+        return $depth;
+    }
+
+    /**
+     * Returns the current leaf child admin instance,
+     * or null if there's no current child.
+     *
+     * @return AdminInterface|null
+     */
+    final public function getCurrentLeafChildAdmin()
+    {
+        $child = $this->getCurrentChildAdmin();
+
+        if (null === $child) {
+            return;
+        }
+
+        for ($c = $child; null !== $c; $c = $child->getCurrentChildAdmin()) {
+            $child = $c;
+        }
+
+        return $child;
+    }
```

Comment:
There should probably be a new interface with these methods, and the interface should either be extended by `AdminInterface`, or directly implemented by `AbstractAdmin` (that would violate the ISP a bit less).

Reply:
> the interface should either be extended by `AdminInterface`.

Not doable on `3.x`. IMO I should do it with a `NEXT_MAJOR` comment.

Reply:
Why?

Reply:
http://symfony.com/doc/current/contributing/code/bc.html#changing-interfaces

Reply:
Oh right, I'm tired!

Reply:
> Type of Change | Change Allowed
> -- | --
> ... | ...
> Add parent interface | Yes *
> ... | ...
> 
> \* The added parent interface must not introduce any new methods that don't exist in the interface already.

Reply:
Obviously

Reply:
> IMO I should do it (commented) with a NEXT_MAJOR comment.

That would go against the ISP though... how about the other solution instead?

Reply:
> how about the other solution instead?

So I create a new interface? This is going to useless if only the `AbstractAdmin` use it, no?

I don't even know how to call it... `AdminTreeInterface`? The `ParentAdminInterface` should be merged with it at some time (sonata v4?).